local-tests/setup/tinny-environment.ts

```diff
@@ -0,0 +1,366 @@
+      );
+    }
+
+    // -- create an empty array to keep track of all keys
+    this.processEnvs.KEY_IN_USE = new Array(
+      this.processEnvs.PRIVATE_KEYS.length
+    ).fill(false);
+
+    // -- setup rpc
+    if (this.network === LIT_TESTNET.LOCALCHAIN) {
+      this.rpc = this.processEnvs.LIT_RPC_URL;
+    } else {
+      this.rpc = this.processEnvs.LIT_OFFICIAL_RPC;
+    }
+  }
+
+  world: Map<string, TinnyPerson> = new Map();
+
+  /**
+   * Retrieves an available private key from a list, marking it as in use and scheduling
+   * its automatic release. If no unused keys are available, it waits for a set interval
+   * before rechecking.
+   *
+   * This function loops until it finds an unused key, marks it, and returns the key with
+   * its index. If all keys are in use, it logs a wait message and pauses before retrying.
+   *
+   * Outputs:
+   * - privateKey: The selected private key.
+   * - index: The index of the selected key.
+   *
+   * Environment variables required:
+   * - KEY_IN_USE: Boolean array indicating key usage.
+   * - PRIVATE_KEYS: Array of key strings.
+   * - TIME_TO_RELEASE_KEY: Milliseconds until a key is automatically released.
+   * - WAIT_FOR_KEY_INTERVAL: Wait time in milliseconds if no keys are free.
+   */
+  async getAvailablePrivateKey(): Promise<{
+    privateKey: string;
+    index: number;
+  }> {
+    while (true) {
+      const index = this.processEnvs.KEY_IN_USE.findIndex((used) => !used); // Find the first unused key
+
+      if (index !== -1) {
+        // If an available key is found
+        this.processEnvs.KEY_IN_USE[index] = true; // Mark the key as in use
+        console.log('[𐬺🧪 Tinny Environment𐬺] 🔑 Selected key at index', index); // Log a message indicating that we have selected a key
+
+        // Set a timer to automatically release the key after 10 seconds
+        setTimeout(() => {
+          this.releasePrivateKey(index);
+          console.log(
+            '[𐬺🧪 Tinny Environment𐬺] 🔓 Automatically released key at index',
+            index,
+            `after ${this.processEnvs.TIME_TO_RELEASE_KEY / 10000} seconds`
+          );
+        }, this.processEnvs.TIME_TO_RELEASE_KEY);
+
+        return { privateKey: this.processEnvs.PRIVATE_KEYS[index], index }; // Return the key and its index
+      } else {
+        console.log('[𐬺🧪 Tinny Environment𐬺] No available keys. Waiting...'); // Log a message indicating that we are waiting
+        // Wait for the specified interval before checking again
+        await new Promise((resolve) =>
+          setTimeout(resolve, this.processEnvs.WAIT_FOR_KEY_INTERVAL)
+        );
+      }
+    }
+  }
+
+  /**
+   * Marks a private key as available again after use.
+   * @param {number} index - The index of the key to mark as available.
+   */
+  releasePrivateKey(index: number) {
+    this.processEnvs.KEY_IN_USE[index] = false;
+    console.log(
+      `[𐬺🧪 Tinny Environment𐬺] 🪽 Released key at index ${index}. Thank you for your service!`
```

Comment:
By storing keys in a hash, we could also log more useful info here (e.g. the key instead of the index) for debugging purposes